Core/include/Acts/Surfaces/Surface.hpp

```diff
@@ -444,6 +421,24 @@ class Surface : public virtual GeometryObject,
   virtual Polyhedron polyhedronRepresentation(const GeometryContext& gctx,
                                               size_t lseg) const = 0;
 
+  /// The derivative of bound track parameters w.r.t. alignment
+  /// parameters of its reference surface (i.e. local frame origin in
+  /// global 3D Cartesian coordinates and its rotation represented with
+  /// extrinsic Euler angles)
+  ///
+  /// @param gctx The current geometry context object, e.g. alignment
+  /// @param derivatives Path length derivatives of the free, nominal
+  /// parameters to help evaluate change of free track parameters caused by
+  /// change of alignment parameters
+  /// @param position The position of the paramters in global
+  /// @param direction The direction of the track
+  ///
+  /// @return Derivative of bound track parameters w.r.t. local frame
+  /// alignment parameters
+  const AlignmentToBoundMatrix alignmentToBoundDerivative(
```

Comment:
I would also call this a Jacobian since it is not just a derivative, but a multidimensional one, e.g. `alignmentToBoundJacobian`.

Reply:
We should also consider taking a `FreeVector` as input instead of separate position and direction to avoid conversions.

Reply:
Different MR, in this one I will only tackle localToGlobal and vice verse